docu/Concepts/BusinessRequirements/UC_Contribution_Messaging.md

```diff
@@ -0,0 +1,212 @@
+
+* Status: eingereicht / in Bearbeitung / bestätigt / abgelehnt
+* Messages vorhanden: ja / nein
+* falls Messages vorhanden, wieviele davon sind ungelesen
+
+### searchContributions for all
+
+Mit diesem Service werden alle Contributions aber keine Messages, aller User gelesen und nach ihrem CreatedAt-Datum zeitlich absteigend sortiert. Es werden weitere evtl. transiente Informationen pro Contribution mit geliefert, um die entsprechenden Ausprägungen im Frontend ansteuern zu können:
+
+* Status: eingereicht / in Bearbeitung / bestätigt / abgelehnt
+* Messages vorhanden: ja / nein
+* falls Messages vorhanden, wieviele davon sind ungelesen
+* User-Info: Vorname, Nachname oder sofern vorhanden dann der Alias
+
+### updateContribution
+
+Über diesen Service kann eine Contribution nur verändert werden, wenn sie im Status "*eingereicht*" oder "*in Bearbeitung*" ist. Dies kann nur der Ersteller der Contribution selbst ausführen. Implizit wird dabei zur Historisierung eine 'HISTORY'-Nachricht mit dem ürsprünglichen Inhalt der Contribution erstellt. In der bearbeiteten Contribution wird neben den geänderten Attributen `Memo `und `Amount `das `created_at`-Datum auf den Zeitpunkt des Updates gesetzt.
+
+### deleteContribution
+
+Mit diesem Service kann eine Contribution im Status "*eingereicht*" bzw. "*in Bearbeitung*" und nur vom Ersteller selbst oder von einem Moderator gelöscht werden. Dabei wird nur ein SoftDelete durchgeführt, indem das `deleted_at`-Datum auf den aktuellen Zeitpunkt und das `deleted_by`-Attribut mit der UserId des Users/Moderators gesetzt wird.
+
+### confirmContribution
+
+Über diesen Service kann der Moderator im AdminInterface eine "*eingereichte*" bzw. eine "*in Bearbeitung*" Contribution bestätigen. Dabei wird implizit das Attribut "`confirmed_by`" auf die UserId des Moderators gesetzt und das Attribut "`confirmed_at`" auf den aktuellen Zeitpunkt. Zusätzlich wird eine Nachricht mit dem übergebenen Begründungstext der Confirmation an die Contribution angehängt.
+
+### denyContribution
+
+Über diesen Service kann der Moderator im AdminInterface eine "*eingereichte*" bzw. ein "*in Bearbeitung*" Contribution ablehnen. Dabei wird implizit das Attribut "`denied_by`" auf die UserId des Moderators, das Attribut "`denied_at`" auf den aktuellen Zeitpunkt gesetzt. Zusätzlich wird eine Nachricht mit dem übergebenen Begründungstext der Ablehnung an die Contribution angehängt.
+
+### searchContributionMessages
+
+Dieser Service liefert zu einer bestimmten Contribution alle gespeicherten Nachrichten chronologisch nach dem `CreatedAt`-Datum absteigend sortiert. Neben dem Nachrichtentext und dem CreatedAt-Datum wird auch der User, der die Nachricht erstellt hat, geliefert. Als User-Daten wird entweder der Vorname und Nachname oder falls vorhanden der Alias geliefert.
```

Comment:
As we have a pagination on the contribution lists, I suggest to deliver all the messages with the contributions. Counts, etc. can be done by the frontend. The question is, do we need pagination for the messages? Do we want a limit of the messages loaded?

Reply:
discussed and clairified during pairing
- yes pagination for contribution-list
- no contributions will be read without messages, but with count of the amount of messages per contribution directly in database
- message reading per additional request and prepared for pagination of messages